From 08e8e9538e898e17e511377aa6092604742d9198 Mon Sep 17 00:00:00 2001
From: Zhang Zhijie <zhangzj@rock-chips.com>
Date: Tue, 5 Mar 2019 18:47:47 +0800
Subject: [PATCH] OPTEE: do uart port config when driver probe

Some logs may appear before tee-supplicant start,
so uart port config should be done as early as possible.

Change-Id: I51bdb6a9d0f5160a6dc66ad015577a77df6897b4
Signed-off-by: Zhang Zhijie <zhangzj@rock-chips.com>
---
 security/optee_linuxdriver/armtz/tee_tz_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/security/optee_linuxdriver/armtz/tee_tz_drv.c b/security/optee_linuxdriver/armtz/tee_tz_drv.c
index fd6e069c2cb0..8a5ad6784820 100644
--- a/security/optee_linuxdriver/armtz/tee_tz_drv.c
+++ b/security/optee_linuxdriver/armtz/tee_tz_drv.c
@@ -1224,8 +1224,6 @@ static int tz_start(struct tee *tee)
 	BUG_ON(ptee->started);
 	ptee->started = true;
 
-	rk_set_uart_port(ptee);
-
 	ret = configure_shm(ptee);
 	if (ret)
 		goto exit;
@@ -1372,6 +1370,8 @@ static int tz_tee_probe(struct platform_device *pdev)
 	if (ret)
 		goto bail1;
 
+	rk_set_uart_port(ptee);
+
 #ifdef _TEE_DEBUG
 	pr_debug("- tee=%p, id=%d, iminor=%d\n", tee, tee->id,
 		 tee->miscdev.minor);
-- 
2.35.3

